src/TheThingsNode.cpp

```diff
@@ -183,54 +186,84 @@ void TheThingsNode::loop()
       }
       this->motionStarted = false;
     }
-    TTN_MOTION_STOP = false;
+    // ACK our interrupt
+    wakeStatus &= ~TTN_WAKE_MOTION_STOP;
   }
 
-  if (TTN_TEMPERATURE)
+  if (wakeStatus & TTN_WAKE_TEMPERATURE)
   {
     if (this->temperatureEnabled && this->temperatureCallback)
     {
       this->temperatureCallback();
     }
-    TTN_TEMPERATURE = false;
+    // ACK our interrupt
+    wakeStatus &= ~TTN_WAKE_TEMPERATURE;
+  }
+
+  if (wakeStatus & TTN_WAKE_LORA)
+  {
+    // This is mainly our interval
+    wakeStatus |= TTN_WAKE_INTERVAL;
+
+    // ACK our interrupt
+    wakeStatus &= ~TTN_WAKE_LORA;
+  }
+
+  if (wakeStatus & TTN_WAKE_WATCHDOG)  
+  {
+    if (TTN_INTERVAL >= this->intervalMs) {
+      wakeStatus |= TTN_WAKE_INTERVAL;
+    }
+    // ACK our interrupt
+    wakeStatus &= ~TTN_WAKE_WATCHDOG;
   }
 
-  if (TTN_INTERVAL >= this->intervalMs)
+  if (wakeStatus & TTN_WAKE_INTERVAL)  
   {
     if (this->intervalEnabled && this->intervalCallback)
     {
-      this->intervalCallback();
+      this->intervalCallback(wakeStatus);
     }
+    // Ack our interval interrupt
+    wakeStatus &= ~TTN_WAKE_INTERVAL;
     TTN_INTERVAL = 0;
   }
 
-  if (this->sleepCallback)
-  {
-    this->sleepCallback();
-  }
+  // If button pushed manage loop faster 
+  uint16_t dly = this->buttonPressed ? 10 : 100;
 
+  // USB is connected and so sleep on USB
   if (this->isUSBConnected() && !this->USBDeepSleep)
   {
-
-    while (!TTN_BUTTON_PRESS && !TTN_BUTTON_RELEASE && !TTN_MOTION_START && !TTN_MOTION_STOP && !TTN_TEMPERATURE && TTN_INTERVAL < this->intervalMs)
+    // Loop until pseudo wake event (because we're not sleeping) or interval
+    while (!(wakeStatus&TTN_WAKE_ANY) && TTN_INTERVAL < this->intervalMs)
```

Comment:
Spacing + consistent order (i.e. constant on right side)

Reply:
Not sure to understand what you mean, would you provide an example?